suffuse/attr.go

```diff
@@ -0,0 +1,76 @@
+package suffuse
+
+import (
+  "os"
+  "time"
+)
+
+/** Eventually I think these types should be code-generated
+ *  so there are custom well-typed attribute containers for each
+ *  type of metadata. For now this is close enough.
+ */
+type AttrKey string
+type AttrValue interface{}
+type AttrMap map[AttrKey]AttrValue
+
+type DirList    map[Name]*Inode
+type Gid        uint32
+type InodeNum   uint64
+type LinkTarget Path
+type Name       string
+type PermBits   os.FileMode
+type Times      [2]time.Time
+type Uid        uint32
+type Umask      os.FileMode
+
+const BytesKey      = AttrKey("Bytes")
+const DirListKey    = AttrKey("DirList")
+const GidKey        = AttrKey("Gid")
+const InodeTypeKey  = AttrKey("InodeType")
+const InodeNumKey   = AttrKey("InodeNum")
+const LinkTargetKey = AttrKey("LinkTarget")
+const NameKey       = AttrKey("Name")
+const PermBitsKey   = AttrKey("PermBits")
+const TimesKey      = AttrKey("Times")
+const UidKey        = AttrKey("Uid")
+
+// man 2 chown:
+//   One of the owner or group id's may be left unchanged by specifying it as -1.
+// Thanks chown, we'll try to forget that uid_t and gid_t are unsigned and pass a
+// bit pattern which should look like -1 after you cast it back to a signed int.
+const MaxValueUint32 = uint32(1 << 32 - 1)
+
+var AllPermBits  = PermBits(os.ModePerm)
+var NoGid        = Gid(MaxValueUint32)
+var NoInodeNum   = InodeNum(0)
+var NoLinkTarget = LinkTarget("")
+var NoName       = Name("")
+var NoUid        = Uid(MaxValueUint32)
+var OsGid        = Gid(os.Getgid())
+var OsUid        = Uid(os.Getuid())
+var OsUmask      = GetUmask()
+
+func (x *Inode) Bytes()Bytes           { return x.AttrOr(BytesKey, NoBytes).(Bytes)                }
```

Comment:
I think we need a formatter (which is very unlikely with the style we type), but in my opinion you should place spaces between function and return type (as you have been doing in earlier commits).